File: feb-05-minimum-removals-to-balance-array/solution.java
import java.util.Arrays;

class Solution {
    public int minRemoval(int[] nums, int k) {
        Arrays.sort(nums);
        int n = nums.length;
        int best = 1;
        int left = 0;
        for (int right = 0; right < n; right++) {
            while ((long) nums[right] > (long) nums[left] * k) {
                left++;
            }
            best = Math.max(best, right - left + 1);
        }
        return n - best;
    }
}

File: feb-05-minimum-removals-to-balance-array/solution.py
from typing import List


class Solution:
    def minRemoval(self, nums: List[int], k: int) -> int:
        nums.sort()
        n = len(nums)
        best = 1
        left = 0
        for right in range(n):
            while nums[right] > nums[left] * k:
                left += 1
            best = max(best, right - left + 1)
        return n - best
